src/Arrays/NSProductOfArrayExceptSelf.java

```diff
@@ -0,0 +1,33 @@
+package Arrays;
+
+/*Given an array nums of n integers where n > 1,  return an array output such that output[i] is equal to the product of all the elements of nums except nums[i].
+
+Example:
+
+Input:  [1,2,3,4]
+Output: [24,12,8,6]
+Constraint: It's guaranteed that the product of the elements of any prefix or suffix of the array (including the whole array) fits in a 32 bit integer.
+
+Note: Please solve it without division and in O(n).
+
+Follow up:
+Could you solve it with constant space complexity? (The output array does not count as extra space for the purpose of space complexity analysis.)*/
+
+class NSProductOfArrayExceptSelf {
+    public int[] productExceptSelf(int[] nums) {
+        int[] op = new int[nums.length];
+        int temp = 1;
+        for(int i = nums.length - 1; i > 0; i--){
+            op[i] = temp*nums[i];
+            temp = op[i];
+        }
+        op[0] = op[1];
```

Comment:
So this statement can be taken inside right. If you alter it to op[i]=op[i+1]*tmp?